src/ShellyEssentials/EventListener.php

```diff
@@ -58,16 +58,18 @@ public function onEntityDamage(EntityDamageEvent $event) : void{
 	}
 
 	public function onPlayerLogin(PlayerLoginEvent $event) : void{
-		$event->getPlayer()->teleport(API::getMainInstance()->getServer()->getDefaultLevel()->getSafeSpawn());
+        $player = $event->getPlayer();
+        API::getMainInstance()->getScheduler()->scheduleDelayedTask(new JoinTitleTask($player), 30);
```

Comment:
This task wont work with the current delay because PlayerLoginEvent triggers way before PlayerJoinEvent, please test and check this...